src/app/page.tsx

```diff
@@ -1,34 +1,78 @@
 "use client";
 
+import { useRouter } from "next/navigation";
 import { useState } from "react";
+import toast from "react-hot-toast";
+import { LOCAL_STORAGE_KEYS } from "@/constants/storage";
 
 export default function AuthPage() {
+  const router = useRouter();
+  const [isLogin, setIsLogin] = useState(false);
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
 
-  const handleSubmit = async (e) => {
+  const handleLogin = async (e: React.FormEvent) => {
     e.preventDefault();
+    try {
+      const res = await fetch("/api/login", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ username, password }),
+      });
 
-    const payload = { username, password }; /// NAMING CONVENTION TO SEND DATA - PAYLOAD/////
-    const res = await fetch("/api/signup", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify(payload),
-    });
+      if (!res.ok) {
+        const errorData = await res.json();
+        toast.error(errorData.error || "Login failed");
+        return;
+      }
 
-    const datareceived = await res.json();
+      const data = await res.json();
```

Comment:
Remove data, if its not being used. or store the id and username in the localstorage.